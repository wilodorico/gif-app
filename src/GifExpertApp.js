import React, { useState } from 'react'
import AddCategory from './components/AddCategory'
import GifGrid from './components/GifGrid'

const GifExpertApp = ({ defaultCategory = []}) => {

  const [category, setCategory] = useState(defaultCategory)

  return (
    <>
      <h2>GifExpertapp</h2>
      <AddCategory setCategory={ setCategory } />
      <hr />

      <ol>
        {category.map((item, index) => (
          <GifGrid key={index} category={item} />
        ))}
      </ol>
    </>
  )
}

export default GifExpertApp
