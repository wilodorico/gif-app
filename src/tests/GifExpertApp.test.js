import React from "react"
import { shallow } from "enzyme"
import GifExpertApp from "../GifExpertApp"


describe('Test component <GifExpertApp />', () => {
   
   test('should display the component correctly', () => { 

      const wrapper = shallow( <GifExpertApp /> )
      expect(wrapper).toMatchSnapshot()

    })

    test('should render list category', () => { 

      const category = ['One punch', 'Dragon Ball']
      const wrapper = shallow( <GifExpertApp defaultCategory={category} /> )

      expect(wrapper).toMatchSnapshot()
      expect(wrapper.find('GifGrid').length).toBe(category.length)

     })



})