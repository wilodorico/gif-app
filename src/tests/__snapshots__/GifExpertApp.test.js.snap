// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component <GifExpertApp /> should display the component correctly 1`] = `
<Fragment>
  <h2>
    GifExpertapp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Test component <GifExpertApp /> should render list category 1`] = `
<Fragment>
  <h2>
    GifExpertapp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="0"
    />
    <GifGrid
      category="Dragon Ball"
      key="1"
    />
  </ol>
</Fragment>
`;
